File: 01 Introdução/1176 Fibonacci em Vetor/1176.py
cases = int(input())
fibs = [0, 1]

for i in range(0, cases):
    x = int(input())

    try:
        fib = fibs[x]
    except:
        quantidade = len(fibs)
        while quantidade != x+1:
            fibs.append(fibs[-1] + fibs[-2])
            quantidade += 1
        fib = fibs[-1]

    print(f"Fib({x}) = {fibs[x]}")

# https://www.beecrowd.com.br/judge/pt/problems/view/1176
# Ranking 05944º, Time 0.127, Submiss Date 09/03/2023 02:55:41 (Python 3.9)

File: 01 Introdução/1176 Fibonacci em Vetor/1176.cpp
#include <iostream>
#include <vector>

int main() {
    int cases;
    std::cin >> cases;

    std::vector<long long> fibs = {0, 1};

    for (int i = 0; i < cases; ++i) {
        int x;
        std::cin >> x;

        try {
            long long fib = fibs.at(x);
        } catch (...) {
            int quantidade = fibs.size();
            while (quantidade != x + 1) {
                fibs.push_back(fibs.back() + fibs[fibs.size() - 2]);
                ++quantidade;
            }
        }

        std::cout << "Fib(" << x << ") = " << fibs[x] << std::endl;
    }

    return 0;
}
